lib/ansible/modules/source_control/github_webhook_facts.py

```diff
@@ -0,0 +1,146 @@
+
+DOCUMENTATION = '''
+---
+module: github_webhook_facts
+short_description: Query information about Github webhooks
+version_added: "2.6"
+description:
+  - "Query information about Github webhooks"
+requirements:
+  - "PyGithub >= 1.3.5"
+options:
+  repository:
+    description:
+      - Full name of the repository to configure a hook for
+    required: true
+  user:
+    description:
+      - User to authenticate to Github as
+    required: true
+  password:
+    description:
+      - Password to authenticate to Github with
+    required: false
+  token:
+    description:
+      - Token to authenticate to Github with
+    required: false
+  github_url:
+    description:
+      - Base URL of the github api
+    required: false
+    default: https://api.github.com
+
+author:
+  - "Chris St. Pierre (@stpierre)"
+'''
+
+EXAMPLES = '''
+# list hooks for a repository (password auth)
+- github_webhook_facts:
+    repository: ansible/ansible
+    user: "{{ github_user }}"
+    password: "{{ github_password }}"
+  register: ansible_webhooks
+
+# list hooks for a repository on github enterprise (token auth)
+- github_webhook_facts:
+    repository: myorg/myrepo
+    user: "{{ github_user }}"
+    token: "{{ github_user_api_token }}"
+    github_url: https://github.example.com/api/v3/
+  register: myrepo_webhooks
+'''
+
+RETURN = '''
+---
+hooks:
+  description: A list of hooks that exist for the repo
+  returned: always
+  type: list
+  sample: >
+    [{"has_shared_secret": true,
+      "url": "https://jenkins.example.com/ghprbhook/",
+      "events": ["issue_comment", "pull_request"],
+      "insecure_ssl": "1",
+      "content_type": "json",
+      "active": true,
+      "id": 6206,
+      "last_response": {"status": "active", "message": "OK", "code": 200}}]
+'''
+
+import traceback
+
+try:
+    import github
+    HAS_GITHUB = True
+except ImportError:
+    HAS_GITHUB = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+
+def _munge_hook(hook_obj):
+    retval = {
+        "active": hook_obj.active,
+        "events": hook_obj.events,
+        "id": hook_obj.id,
+        "url": hook_obj.url,
+    }
+    retval.update(hook_obj.config)
+    retval["has_shared_secret"] = "secret" in retval
+    if "secret" in retval:
+        del retval["secret"]
+
+    retval["last_response"] = hook_obj.last_response.raw_data
+    return retval
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            repository=dict(type='str', required=True),
+            user=dict(type='str', required=True),
+            password=dict(type='str', required=False, no_log=True),
+            token=dict(type='str', required=False, no_log=True),
+            github_url=dict(
+                type='str', required=False, default="https://api.github.com")),
+        mutually_exclusive=(('password', 'token'), ),
+        required_one_of=(("password", "token"), ),
+        supports_check_mode=True)
```

Comment:
I do not think so `check mode` is required for facts module.

Reply:
What does this mean? Why would you not want facts during a check mode run?

Reply:
This flag denotes that module will not perform any changes to remote system if `--check` is provided while running. Fact modules does not change anything on remote systems so it is pointless to have `supports_check_mode` as True. Moreover, there is no action performed with respect to `module.check_mode` like done [here](https://github.com/ansible/ansible/blob/2ad7d79985fef9abc2ee1c5166762b5773cf5c4d/lib/ansible/modules/cloud/vmware/vmware_guest.py#L2095).

Reply:
If a module does not support check mode (i.e., `supports_check_mode=False`), then it will be skipped when run in check mode. If we skip a facts module in check mode, then it could significantly impact the run -- facts could be missing that many tasks depend on, for instance. Additionally, since this module *never* changes anything, it doesn't need to do anything special for check mode.

So yes, this module *does* support check mode: it always runs in a manner that doesn't change anything, but only predicts if anything would be changed (spoiler alert: it wouldn't). And moreover, changing this line to `False` could have unnecessary negative effects if someone were to depend on facts from this module that would then become unavailable in check mode _for no good reason_.

Reply:
Make sense to me. Thanks for info.